### DISCUSSION TITLE
How do I add a valuebox as a tab in a dashboard?

### DISCUSSION THREAD

**Discussion by pensivedog** [quarto-dev/quarto-cli#11374] [Q&A]
### Description

Can anyone provide a working example of this? I've tried a number of approaches following documentation on tabset and valueboxes and can't get them to work tegether. I'm using Quarto 1.5.57.

**Comment by mcanouil** [comment]
It would help if you showed what you tried.

**Comment by pensivedog** [comment]
Thanks for the reply. I have tried the below: 

````
## Row

### Column {.tabset}

```{r}
#| title: "Title Here"
plot_here
```

::: {.valuebox icon="cloud-sun" color="secondary" title="Valuebox Title Here"}
`r value_goes_here`
:::

### Column
```{r}
#| title: "Plot Title"
plot
```
````

I also tried with the other valuebox syntax:

````
```{r}
#| content: valuebox
#| title: "Spam per day"
list(
  icon = "trash",
  color = "danger",
  value = spam
)
```

````

No luck. Help appreciated!

**Reply by mcanouil** [reply]
Could you properly format your post using code blocks for code and terminal outputs? Thanks.
If your code contains code blocks, you need to enclose it using more backticks, *i.e.*, usually four ` ```` `.
See <https://quarto.org/bug-reports.html#formatting-make-githubs-markdown-work-for-us>.

**Reply by pensivedog** [reply]
Thanks for explaining. Updated!

**Comment by mcanouil** [comment]
I am not sure value box was intended to be used like that but maybe it should work without workaround (cc @cscheid)

If you add an extra fenced div, it works as expected.

<table>
<tr><th>Input</th><th>Output</th></tr>
<tr><td>

````qmd
---
title: "Quarto Playground"
format: dashboard
---

## Row

### Column {.tabset}

```{r}
#| title: "Title Here"
plot(1)
```

:::: {}
::: {.valuebox icon="cloud-sun" color="secondary" title="Valuebox Title Here"}
`{r} 42`
:::
::::

### Column

```{r}
#| title: "Plot Title"
plot(1)
```

````

</td><td>

<img width="729" alt="image" src="https://github.com/user-attachments/assets/b771acc9-a664-4790-8c22-8b6ef0284e40">

</td></tr>
</table>

For reference, without the extra fenced div, it is rendered as in the screenshot below, *i.e.*, everything that is not text are removed

<img width="726" alt="image" src="https://github.com/user-attachments/assets/fd5acaf5-75bb-4361-b265-551a5eb73bf8">

**Reply by pensivedog** [reply]
Brilliant. That indeed works. Thank you! Only trouble now is how to change the tab title from "Tab 2". Any ideas for that?

**Reply by mcanouil** [reply]
The title is define by the header.

So better that the extra fenced div to ensure value box is not top-level, is to use the header.

Now, I realised, I don't think there is actually a bug or an enhancement.

````qmd
---
title: "Quarto Playground"
format: dashboard
---

## Row

### Column {.tabset}

```{r}
#| title: "Title Here"
plot(1)
```

#### My tab header

::: {.valuebox icon="cloud-sun" color="secondary" title="Valuebox Title Here"}
`{r} 42`
:::

### Column

```{r}
#| title: "Plot Title"
plot(1)
```
````

**Reply by pensivedog** [reply]
Ok, this is getting closer! Now, the issue is that having third tab in this `.tabset` column breaks the row/column layout.

**Reply by mcanouil** [reply]
Now you need to follow the documentation.
<img width="468" alt="image" src="https://github.com/user-attachments/assets/ecf26b96-a4dc-40ff-8cff-b665cc99e874">

**Reply by pensivedog** [reply]
Alright. Thanks for your help!

**Reply by pensivedog** [reply]
Ok, here's what I mean. Try adding a second identical row to this example:

````
---
title: "Quarto Playground"
format: dashboard
scrolling: true
---

## Row

### Column {.tabset}

```{r}
#| title: "Title Here"
plot(1)
```

#### My tab header

::: {.valuebox icon="cloud-sun" color="secondary" title="Valuebox Title Here"}
`{r} 42`
:::

#### Another tab

::: {.valuebox icon="cloud-sun" color="secondary" title="Valuebox Title Here"}
`{r} 42`
:::

### Column

```{r}
#| title: "Plot Title"
plot(1)
```

## Row

### Column {.tabset}

```{r}
#| title: "Title Here"
plot(1)
```

#### My tab header

::: {.valuebox icon="cloud-sun" color="secondary" title="Valuebox Title Here"}
`{r} 42`
:::

#### Another tab

::: {.valuebox icon="cloud-sun" color="secondary" title="Valuebox Title Here"}
`{r} 42`
:::

### Column

```{r}
#| title: "Plot Title"
plot(1)
```

````

The result is a broken row/column layout:

![Screenshot 2024-11-13 120937](https://github.com/user-attachments/assets/86816ade-b1fc-403b-b74d-8d14c5c42ce3)

**Comment by mcanouil** [comment]
Thanks.

<table>
<tr><th>Input</th><th>Output</th></tr>
<tr><td>

````qmd
---
title: "Quarto Playground"
format: dashboard
---

## Row

### Column {.tabset}

#### Tab 1

{{< lipsum 1 >}}

#### Tab 2

{{< lipsum 1 >}}

### Column

{{< placeholder >}}

## Row

### Column {.tabset}

#### Tab 1

{{< lipsum 1 >}}

#### Tab 2

{{< lipsum 1 >}}

### Column

{{< placeholder >}}

````

</td><td>

<img width="726" alt="image" src="https://github.com/user-attachments/assets/751da71b-978d-40c3-b36e-43fd1caeaec8">

</td></tr>
</table>

**Reply by mcanouil** [reply]
I've opened an issue to track this:

- https://github.com/quarto-dev/quarto-cli/issues/11387

**Reply by pensivedog** [reply]
It seems it's not `.tabset` that's causing the problem but the `.valuebox`. See the below in which `.tabset` works and does not break layouts:

````
---
title: "Quarto Playground"
format: dashboard
scrolling: true
---

## Row

### Column {.tabset}

```{r}
#| title: "Title Here"
plot(1)
```

::: {.card title="tab 2"}
`{r} 42`
:::

::: {.card title="tab 3"}
`{r} 42`
:::

### Column

```{r}
#| title: "Plot Title"
plot(1)
```

## Row

### Column {.tabset}

```{r}
#| title: "Title Here"
plot(1)
```

::: {.card}
`{r} 42`
:::


::: {.card}
`{r} 42`
:::

### Column

```{r}
#| title: "Plot Title"
plot(1)
```

````

![image](https://github.com/user-attachments/assets/af025b34-b070-40c5-9ac7-486d534a682e)

**Reply by mcanouil** [reply]
Yes, that's not the value box.
See my example.

**Reply by pensivedog** [reply]
Ah, right I see. Seems to be something about how headings are defined.

**Reply by mcanouil** [reply]
A bit more on this, from <https://github.com/quarto-dev/quarto-cli/issues/11387#issuecomment-2473320119>:

The issue/confusion here comes from the difference with standard `HTML` formats.
Contrary to those, `dashboard` may seem to work by defining tab panels using explicit headers but it actually does not.
Instead, `.card` should be used.

So, it might not be considered a bug per se, but very likely to bit many users as is.

<table>
<tr><th>Input</th><th>Output</th></tr>
<tr><td>

````qmd
---
title: "Quarto Playground"
format: dashboard
---

## Row

### Column {.tabset}

::: {.card title="Tab 1"}
{{< lipsum 1 >}}
:::

::: {.card title="Tab 2"}
{{< lipsum 1 >}}
:::

### Column

{{< placeholder >}}

## Row

### Column {.tabset}

::: {.card title="Tab 1"}
{{< lipsum 1 >}}
:::

::: {.card title="Tab 2"}
{{< lipsum 1 >}}
:::

### Column

{{< placeholder >}}

````

</td><td>

![Image](https://github.com/user-attachments/assets/35220225-826a-4cc6-baa3-6408fadb46fc)

</td></tr>
</table>

**Reply by pensivedog** [reply]
I see. But is it possible to put a valuebox inside a tab defined by `.card`?

**Reply by mcanouil** [reply]
Probably not, as `cards` are for markdown.

But, in my opinion, it's weird to have a value box as a panel in a tabset.
Value boxes are usually top-level components in dashboards.

**Reply by pensivedog** [reply]
Then it seems valueboxes don't work in tabsets in dashboards, as is suggested in #7405?

I agree, it's a bit of an edge case. We're using them as top-level components as well, but have some cases where we want to have them in tabsets. We can find another way to do it, just wanted to see if this was possible, as it would be better than a custom html/css workaround.

**Reply by mcanouil** [reply]
I have to say that I don't know since no PRs or commits were attached.
It would need further investigation to check if it did work on 1.4.

**Reply by pensivedog** [reply]
Ok, very much appreciate your efforts on this!

**Reply by mcanouil** [reply]
To summarise this thread a bit:

1. value box does not seem to work fully in tabset (https://github.com/quarto-dev/quarto-cli/issues/7405)
2. tabset breaks row/column layout when using header to define new tab.

**Reply by mcanouil** [reply]
Here is a full depiction of the issues when nesting/combining value boxes and tabset

````qmd
---
title: "Quarto Playground"
format: dashboard
---

# Cards

## Row

### Column {.tabset}

::: {.card title="Tab 1"}
::: {.valuebox icon="pencil" color="blue"}
Articles per day

42
:::
:::

::: {.card title="Tab 2"}
::: {.valuebox icon="pencil" color="blue"}
Articles per day

42
:::
:::

### Column

{{< placeholder >}}

## Row

### Column {.tabset}

#### Tab 1

::: {.valuebox icon="pencil" color="blue"}
Articles per day

42
:::

#### Tab 2

::: {.valuebox icon="pencil" color="blue"}
Articles per day

42
:::

### Column

{{< placeholder >}}

# Headers

## Row

### Column {.tabset}

#### Tab 1

::: {.valuebox icon="pencil" color="blue"}
Articles per day

42
:::

#### Tab 2

::: {.valuebox icon="pencil" color="blue"}
Articles per day

42
:::

### Column

{{< placeholder >}}

## Row

### Column {.tabset}

::: {.card title="Tab 1"}
::: {.valuebox icon="pencil" color="blue"}
Articles per day

42
:::
:::

::: {.card title="Tab 2"}
::: {.valuebox icon="pencil" color="blue"}
Articles per day

42
:::
:::

### Column

{{< placeholder >}}

````

<img width="1233" alt="image" src="https://github.com/user-attachments/assets/7b4e76a9-63cf-4349-a2cd-d02f5b03d749">
<img width="1238" alt="image" src="https://github.com/user-attachments/assets/c8fb122e-7716-489f-8c80-65c8fc4c6e55">

Both rows and pages should display properly the value boxes and the two column layout per rows.
